concept/impl/ConceptImpl.java

```diff
@@ -39,12 +39,11 @@
  */
 public abstract class ConceptImpl implements Concept, ConceptVertex {
     private final VertexElement vertexElement;
-    final ConceptManagerImpl conceptManager;
+    ConceptManagerImpl conceptManager;
```

Comment:
CAUSE LAMBDA

Reply:
what lambda

Reply:
ah I see, but `final` carries a meaning when used as a field - it tells us the it will passed as a collaborator in the constructor. So I would try to find a workaround for the lambda rather than sacrificing the `final`

Reply:
and also the `final` here makes actual sense, so that nobody tries to override it

Reply:
can remove the cache and use a simple null initialiser

Reply:
can we initialise the caches in the constructor?

Reply:
can try that as well

Reply:
yeah, I think initialising things outside the constructor makes only sense if `static`